crates/runtime/src/datafusion.rs

```diff
@@ -294,6 +341,23 @@ impl DataFusion {
             .await
             .context(UnableToGetTableSnafu)?;
 
+        Ok(table_provider)
+    }
+
+    pub async fn write_data(
+        &self,
+        table_reference: TableReference,
+        data_update: DataUpdate,
+    ) -> Result<()> {
+        if !self.is_writable(&table_reference) {
+            TableNotWritableSnafu {
+                table_name: table_reference.to_string(),
+            }
+            .fail()?;
+        }
+
+        let table_provider = self.get_table_provider(table_reference.clone()).await?;
```

Comment:
Maybe this can take a reference so we don't have to clone?

Reply:
fixed, thanks